Sources/TensorFlow/Operators/Math.swift

```diff
@@ -2158,6 +2158,36 @@ internal func _vjpMatmul<Scalar: TensorFlowFloatingPoint>(
     })
 }
 
+public extension Tensor where Scalar: TensorFlowFloatingPoint {
+    /// Computes the QR decomposition of each inner matrix in `tensor` such that
+    /// `tensor[..., :, :] = q[..., :, :] * r[..., :,:])`
+    /// 
+    /// - Parameters:
+    ///   - fullMatrices: If true, compute full-sized `q` and `r`. If false
+    ///     (the default), compute only the leading `P` columns of `q`.
+    ///  
+    func qr(fullMatrices: Bool = false) -> (Tensor<Scalar>, Tensor<Scalar>) {
+        return Raw.qr(self, fullMatrices: fullMatrices)
+    }
+
+    /// Returns the diagonal part of the tensor.
+    ///
+    /// For example:
+    ///
+    /// ```
+    /// # 't' is [[1, 0, 0, 0]
```

Comment:
Use Swift code in code snippets. Swift comments start with `//`.
```suggestion
    /// // 't' is [[1, 0, 0, 0]
```